backend/src/apiserver/client_manager/client_manager.go

```diff
@@ -178,6 +198,28 @@ func (c *ClientManager) init() {
 	// UUID generator
 	c.uuid = util.NewUUIDGenerator()
 
+	var controllerClient ctrlclient.Client
+
+	if useKubernetesWebhookMode {
```

Comment:
Sorry, I think the design document was a little confusing, but the intent here is that the normal case is that the API server would *also* serve the webhooks.

In the event that you have multiple Kubeflow Pipelines installations on the same cluster (e.g. Open Data Hub), you may want just a single webhook for all Kubeflow Pipeline installations on the cluster. This is okay because the webhooks don't require state beyond what's in Kubernetes.

So to summarize, `useKubernetesWebhookMode` should be `onlyKubernetesWebhookMode` so that the two options are "webhook + normal REST API endpoints" or "just webhook".